superset-frontend/src/explore/exploreUtils.js

```diff
@@ -288,8 +288,9 @@ export const exploreChart = formData => {
   postForm(url, formData);
 };
 
-export const useDebouncedEffect = (effect, delay) => {
-  const callback = useCallback(effect, [effect]);
+export const useDebouncedEffect = (effect, delay, deps) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const callback = useCallback(effect, deps);
```

Comment:
```suggestion
  const callback = useCallback(effect, [effect, deps]);
```

Reply:
This will cause the issue to happen again unofrtunately